File: WebApp/Scripts/App/Book.js
(function (libApp) {
    function book() {
        var self = this;

        self.Id = ko.observable("");

        self.Name = ko.observable("");

        self.Author = ko.observable("");

        self.PublishedDate = ko.observable("");
        
        self.BookShelfId = ko.observable("");
    }
    libApp.model.Book = book;
}(window.LibApp));

File: Library.Model/Models/Book.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Library.Model.Models
{
    public class Book: BaseEntity
    {
        public string Name { set; get; }

        public string Author { set; get; }

        public DateTime PublishDate { set; get; }

        public BookShelf BookShelf { set; get; }

        public User Reader { set; get; }
    }
}
